compiler/src/dotty/tools/dotc/util/WeakHashSet.scala

```diff
@@ -129,77 +135,71 @@ abstract class WeakHashSet[A <: AnyRef](initialCapacity: Int = 8, loadFactor: Do
     tableLoop(0)
   }
 
-  def lookup(elem: A): A | Null = elem match {
-    case null => throw new NullPointerException("WeakHashSet cannot hold nulls")
```

Comment:
why was this changed? I think it's better if WeakHashSets don't contains nulls.

Reply:
Reverted